lib/connectors/example/simple/connector.rb

```diff
@@ -0,0 +1,98 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+
+# frozen_string_literal: true
+
+require 'connectors/base/connector'
+require 'core/filtering/validation_status'
+require 'utility'
+
+module Connectors
+  module Example
+    module Simple
+      class Connector < Connectors::Base::Connector
+        def self.service_type
+          'example'
+        end
+
+        def self.display_name
+          'Example Connector'
+        end
+
+        # Field 'Foo' won't have a default value. Field 'Bar' will have the default value 'Value'.
+        def self.configurable_fields
+          {
+            'foo' => {
+              'label' => 'Foo',
+              'value' => nil
+            },
+            :bar => {
+              :label => 'Bar',
+              :value => 'Value'
+            }
+          }
+        end
+
+        def initialize(configuration: {}, job_description: {})
+          super
+        end
+
+        def do_health_check
+          # Do the health check by trying to access 3rd-party system just to verify that everything is set up properly.
+          #
+          # To emulate unhealthy 3rd-party system situation, uncomment the following line:
+          # raise 'something went wrong'
+        end
+
+        def self.validate_filtering(filtering = {})
+          # TODO: real filtering validation will follow later
```

Comment:
We've changed this logic to only be present in the base connector and only override `self.advanced_snippet_validator` (so we plugin a dedicated validation class). 

Just as a heads up, when it's merged :-)